server/service/src/purchase_order_line/query.rs

```diff
@@ -40,3 +40,87 @@ pub fn get_purchase_order_line(
 
     Ok(repository.query_by_filter(filter)?.pop())
 }
+
+#[cfg(test)]
+mod test {
+    use crate::service_provider::ServiceProvider;
+    use repository::mock::mock_store_a;
+    use repository::{db_diesel::PurchaseOrderLineRow, mock::MockDataInserts, test_db::setup_all};
+    use repository::{
+        EqualFilter, PurchaseOrderLineFilter, PurchaseOrderLineRowRepository, PurchaseOrderRow,
+        PurchaseOrderRowRepository,
+    };
+    use util::inline_init;
+    #[actix_rt::test]
+    async fn purchase_order_service_queries() {
+        let (_, connection, connection_manager, _) = setup_all(
+            "purchase order line service queries",
+            MockDataInserts::none(),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager);
+        let context = service_provider.basic_context().unwrap();
+        let service = service_provider.purchase_order_line_service;
+        let repo = PurchaseOrderLineRowRepository::new(&connection);
+
+        // add purchase order
+        let purchase_order_repo = PurchaseOrderRowRepository::new(&connection);
+        let purchase_order_id = "test_po_1";
+        let po = inline_init(|p: &mut PurchaseOrderRow| {
+            p.id = purchase_order_id.to_string();
+            p.store_id = mock_store_a().id;
+        });
+        purchase_order_repo.upsert_one(&po).unwrap();
+
+        let result = purchase_order_repo.find_all().unwrap();
+        assert!(result.is_empty());
+
+        let po = inline_init(|p: &mut PurchaseOrderLineRow| {
+            p.id = "test_po_1".to_string();
+            p.purchase_order_id = purchase_order_id.to_string();
+        });
+        repo.upsert_one(&po).unwrap();
+
+        // Test querying by ID
+        let result = service
+            .get_purchase_order_line(&context, "wrong_store_id", &po.id)
+            .unwrap();
+        assert!(result.is_none());
+
+        let result = service
+            .get_purchase_order_line(&context, &mock_store_a().id, "wrong_line_id")
+            .unwrap();
+        assert!(result.is_none());
+
+        let ref result = service
+            .get_purchase_order_line(&context, &mock_store_a().id, &po.id)
+            .unwrap();
+        assert!(result.is_some());
```

Comment:
This isn't exhaustive testing - but thought it would be helpful to test filtering by store_id as this is via a join